### DISCUSSION TITLE
`sea-orm-cli generate entity` with multiple Postgres schemas?

### DISCUSSION THREAD

**Discussion by nick-kang** [SeaQL/sea-orm#1444] [Q&A]
How do you handle code generation with multiple Postgres schemas with foreign keys in different schemas? We use multiple schemas to organize our tables.

I tried two methods that don't seem to work.

I tried running the cli command twice, but this lead to `mod.rs` only containing the tables for the last schema.

```bash
sea-orm-cli generate entity -s app
sea-orm-cli generate entity -s iam
```

I then tried creating a separate folder for each schema, but the code won't compile because it expected all the entities to be in the same folder (ie, the table referenced another table in a separate schema which was in a separate folder now)

```bash
sea-orm-cli generate entity -o schema/app -s app
sea-orm-cli generate entity -o schema/iam -s iam
```

Our current workaround is to put all tables in one schema, which isn't ideal.

**Comment by billy1624** [comment]
Hey @nick-kang, it should works just fine. Please check that your generated entity definition should have a `schema_name` attribute, just like:

```rs
#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
#[sea_orm(schema_name = "schema_name", table_name = "_cake_filling_")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false)]
    pub cake_id: i32,
    #[sea_orm(primary_key, auto_increment = false)]
    pub filling_id: i32,
}
```
